### ISSUE TITLE
Idea: DrawSprite function that takes a normal vector instead of rotation as an argument.

### ISSUE THREAD

**Issue by s-m-k** [MonoGame/MonoGame#5870]
SpriteBatch.DrawSprite should be overloaded to take an unit vector instead of the rotation angle.

You don't always need to represent a rotation with an angle, but an unit vector instead, which is, in fact, [cos, sin] of the angle. By passing it to DrawSprite directly instead of that angle, you don't need to use Math.Atan2, Math.Cos and Math.Sin each sprite like you do now (to retrieve the angle from [cos, sin] then convert it back to [cos, sin] to perform the rotation).

It should be a huge performance boost when you have lots of moving sprites and unit vectors instead of rotation angles. Now I have to create my own little sprite batch that features it to achieve better performance.

**Comment by nkast** [comment]
> [...], but an unit vector instead, which is, in fact, [cos, sin] of the angle. 

Another name for that is a [Complex number](https://github.com/tainicom/Aether.Physics2D/blob/247c19027978c8950210171ecfa5d70638d397b7/Physics2D/Maths/Complex.cs), but in principle it's the same thing.
It's like a 2x2 rotation matrix, but you don't store the duplicate +cos and -sin.

**Comment by s-m-k** [comment]
Yes, in fact passing the rotation matrix should be possible too.

And yes, I'm using the unit vector often because of complex number multiplication which lets me avoid lots of trig function calls (and also makes the code cleaner).

**Comment by willmotil** [comment]
For a 2d function you can simply wrap your own spritebatch call and use Math.Atan2(y, x).

public static void MyDraw(SpriteBatch sb, Texture2D t, ... ect... , Vector2 rot, ...)
{
   float r = (float)(Math.Atan2( rot.Y, rot.Y);
   // then call the spritebatch.Draw and pass r for rotation. 
}

   however... Math.Atan2 uses a polar axis system so you have to pass like - rot.Y or something i forget.
   For 3d this isn't really proper its non standardized as x,y,z angles can have up to 16 different meanings, 
   depending or the order of mathematical operations performed on the rotations.
   In 2d the axis is standardly defined as being on the z axis.
   So a full rotation matrix would have to be passed for 3d orientations.

**Comment by s-m-k** [comment]
> For a 2d function you can simply wrap your own spritebatch call and use Math.Atan2(y, x).

Which is totally not the point. It's obvious that I can do it, but simply using Atan2 makes me in fact do 3 trigonometric function calls that convert the vector to angle (atan2), then that angle back to the unit vector/complex number (cos, sin) that represents the rotation. By allowing the SpriteBatch to be able to receive the rotation in an unit vector/complex number form or a rotation matrix, the trig functions are not needed. Which means less redundant CPU overhead.

>For 3d this isn't proper to pass a vector3 for rotations its non standardized as x,y,z angles can have up to 16 different meanings, depending or the order of mathematical operations performed on the rotations.

I think it's unrelated. SpriteBatch seems to be focused on 2D. Anyway, in case of 3D, you'd likely analogously pass an unit quaternion (which is just a "higher order" complex number that could be represented by Vector4) instead of an unit complex number (represented by Vector2 in my case).

**Comment by willmotil** [comment]
Well you have a point.
You can drop spritebatch and draw quads directly you will have to pass a full rotation matrix.  
I would actually like to see SpriteBatch made public so people could just make their own extension methods for it.
